src/clients/api/queries/useGetPrimeEstimation/index.ts

```diff
@@ -0,0 +1,173 @@
+  borrowedTokens: BigNumber | undefined;
+  borrowApyPercentage: BigNumber | undefined;
+  borrowCapTokens: BigNumber | undefined;
+  borrowCapCents: BigNumber | undefined;
+  suppliedTokens: BigNumber | undefined;
+  supplyApyPercentage: BigNumber | undefined;
+  supplyCapTokens: BigNumber | undefined;
+  supplyCapCents: BigNumber | undefined;
+  userDailyPrimeRewards: BigNumber | undefined;
+}
+
+export type UseGetPrimeEstimationQueryKey = [
+  FunctionKey.GET_PRIME_ESTIMATION,
+  UseGetPrimeEstimationInput,
+];
+
+type Options = QueryObserverOptions<
+  GetHypotheticalPrimeApysOutput,
+  Error,
+  GetHypotheticalPrimeApysOutput,
+  GetHypotheticalPrimeApysOutput,
+  UseGetPrimeEstimationQueryKey
+>;
+
+const useGetPrimeEstimation = (
+  {
+    accountAddress,
+    borrowedAmountMantissa,
+    stakedAmountXvsMantissa,
+    suppliedAmountMantissa,
+    vToken,
+  }: UseGetPrimeEstimationInput,
+  options?: Options,
+) => {
+  const xvs = useGetToken({
+    symbol: 'XVS',
+  });
+
+  const enabled = !!options?.enabled && !!vToken;
+
+  const { data: primeDistributionForMarketData } = useGetPrimeDistributionForMarket(
+    {
+      vTokenAddress: vToken?.address || '',
+    },
+    {
+      enabled,
+    },
+  );
+
+  const { data: hypotheticalPrimeApysData } = useGetHypotheticalPrimeApys(
+    {
+      vTokenAddress: vToken?.address || '',
+      userSupplyBalanceMantissa: suppliedAmountMantissa,
+      userBorrowBalanceMantissa: borrowedAmountMantissa,
+      userXvsStakedMantissa: stakedAmountXvsMantissa,
+      accountAddress: accountAddress || NULL_ADDRESS,
+    },
+    {
+      enabled,
+    },
+  );
+
+  const estimation = useMemo(() => {
+    let primeEstimation: UseGetPrimeEstimationOutput = {
+      dailyTokensDistributedAmount: undefined,
+      borrowedTokens: undefined,
+      borrowApyPercentage: undefined,
+      borrowCapTokens: undefined,
+      borrowCapCents: undefined,
+      suppliedTokens: undefined,
+      supplyApyPercentage: undefined,
+      supplyCapTokens: undefined,
+      supplyCapCents: undefined,
+      userDailyPrimeRewards: undefined,
+    };
+
+    if (hypotheticalPrimeApysData && primeDistributionForMarketData && vToken && xvs) {
+      const {
+        borrowApyPercentage,
+        borrowCapMantissa,
+        borrowCapCents,
+        supplyApyPercentage,
+        supplyCapMantissa,
+        supplyCapCents,
+        userPrimeRewardsShare,
+      } = hypotheticalPrimeApysData;
+
+      const borrowedTokens = convertMantissaToTokens({
+        value: borrowedAmountMantissa,
+        token: vToken.underlyingToken,
+      });
+
+      const suppliedTokens = convertMantissaToTokens({
+        value: suppliedAmountMantissa,
+        token: vToken.underlyingToken,
+      });
+
+      const borrowCapTokens = convertMantissaToTokens({
+        value: new BigNumber(borrowCapMantissa.toString()),
+        token: vToken.underlyingToken,
+      });
+
+      const supplyCapTokens = convertMantissaToTokens({
```

Comment:
Queries should return mantissas for easy calculation downstream. Otherwise it is probable they'll need to be converted back to Mantissa. Converting to tokens should be done in a view before presenting the value, ideally in a component dedicated to this with all the formatting logic